src/ethereum/frontier/vm/instructions/comparison.py

```diff
@@ -0,0 +1,172 @@
+"""
+Ethereum Virtual Machine (EVM) Comparison Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM Comparison instructions.
+"""
+
+from ethereum.base_types import U256
+
+from .. import Evm
+from ..gas import GAS_VERY_LOW, subtract_gas
+from ..stack import pop, push
+
+
+def lt(evm: Evm) -> None:
+    """
+    Checks if the top element is less than the next top element. Pushes the
+    result back on the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+
+    x = pop(evm.stack)
+    y = pop(evm.stack)
+    result = U256(x < y)
```

Comment:
I wonder if something like this is a bit clearer: 

```python
result = U256(1) if x < y else U256(0)
```

Reply:
I guess not, rest of the comparisons do look nice. I just worry about things not being super explicit.

Reply:
I think it's common to understand that a bool can be converted to `int` or `Uint`. Let me know your thoughts.